src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,121 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFile(statusFilePath, reusableReader, out string fileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            var statusFileContents = fileContents.AsSpan();
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                // Reached end of file
+                if (startIndex == -1)
+                {
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                statusFileContents = statusFileContents.Slice(startIndex + 1);
+                int endIndex = statusFileContents.IndexOf('\n');
+                if (endIndex == -1)
+                {
+                    endIndex = statusFileContents.Length - 1;
+                }
+
+                ReadOnlySpan<char> value = default;
+                int sliceLength = endIndex  - 3;
+                bool valueParsed;
+#if DEBUG
+                if (title.SequenceEqual("Pid".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, endIndex);
+                    valueParsed = int.TryParse(value, out results.Pid);
+                    Debug.Assert(valueParsed);
+                    statusFileContents = statusFileContents.Slice(endIndex + 1);
+                    continue;
+                }
+#endif
+                if (title.SequenceEqual("VmHWM".AsSpan()))
```

Comment:
```suggestion

                int sliceLength = endIndex - 3;
                if (title.SequenceEqual("VmHWM".AsSpan()))
```

Reply:
Why move inside ?

Reply:
You mean why move the sliceLength variable down?  Because it's not used by the code above it, which is its own isolated if block, only for DEBUG, etc.

Reply:
My bad. Viewed it in a different way thinking it moved inside the if statement.

Reply:
Done.